### PR TITLE
fix: improved elastic search query filter

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

This PR contains following fixes and improves:

**Fixes:**

- Fixed ScientificMetadata condition filter. Before the change, filter does not work with floating number, due to incorrect mapping settings for elastic search.

**Improves:**

- Included fallback for elastic search. Before the change, the datasets table is empty if ES DB is not synched with mongoDB. Now, the dataset full query endpoint checks if ES query returns any result, if ES query does not return result then it uses regular mongo query.

## Motivation

Background on use case, changes needed

## Fixes:

* Items added

## Changes:

* changes made

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?

### CODE REVIEW

**Review by bpedersen2 — CHANGES_REQUESTED** [review]
Except for some  minor comments, looks fine.

**Comment by Junjiequan** [comment]
@bpedersen2  Could you copy paste the comments here? I'm not able to see the comments somehow

File: src/elastic-search/configuration/indexSetting.ts [hunk 1]
```diff
@@ -39,12 +39,12 @@ export const dynamic_template:
       path_match: "scientificMetadata.*.*",
```
Comment (bpedersen2) [inline src/elastic-search/configuration/indexSetting.ts:37]: long_as_double??

**Review by bpedersen2 — APPROVED** [review]

**Comment by bpedersen2** [comment]
needs local rebase and force-pushing to `SWAP-3608-scicat-datasets-table-total-number-and-actual-i` branch

**Pull request merged**